contracts/TokenHub.sol

```diff
@@ -522,6 +522,31 @@ contract TokenHub is ITokenHub, System, IParamSubscriber, IApplication, ISystemR
     return elements.encodeList();
   }
 
+  /**
+   * @dev request a BC token aridrop transfer from BSC
+   *
+   * @param contractAddr The token contract which is transferred
+   * @param recipient The destination address of the transfer on BSC.
+   * @param amount The amount to transfer
+   */
+  function unlock(address contractAddr, address recipient, uint256 amount) external override onlyInit onlyAirDrop payable {
+    uint256 convertedAmount;
+    if (contractAddr != address(0x00)) {
+      require(contractAddrToBEP2Symbol[contractAddr] != bytes32(0x00), "not bound");
+      uint256 bep20TokenDecimals=bep20ContractDecimals[contractAddr];
+      convertedAmount = convertToBep2Amount(amount, bep20TokenDecimals);// convert to bep2 amount
+      require(bep20TokenDecimals>=BEP2_TOKEN_DECIMALS || (bep20TokenDecimals<BEP2_TOKEN_DECIMALS && convertedAmount>amount), "amount is too large, uint256 overflow");
+      require(convertedAmount<=MAX_BEP2_TOTAL_SUPPLY, "amount is too large, exceed maximum bep2 token amount");
+      require(IBEP20(contractAddr).balanceOf(address(this)) >= convertedAmount, "insufficient balance");
+      IBEP20(contractAddr).transfer(recipient, amount);
```

Comment:
540,541
why check convertedAmount but use amount to transfer?

Reply:
fixed